fe/fe-core/src/main/java/com/starrocks/planner/ExchangeNode.java

```diff
@@ -92,7 +93,16 @@ public ExchangeNode(PlanNodeId id, PlanNode inputNode, boolean copyConjuncts) {
         }
         // Only apply the limit at the receiver if there are multiple senders.
         if (inputNode.getFragment().isPartitioned()) {
-            limit = inputNode.limit;
+            if (inputNode instanceof SortNode) {
+                SortNode sortNode = (SortNode) inputNode;
+                if (CollectionUtils.isEmpty(sortNode.getSortInfo().getPartitionExprs())) {
+                    limit = inputNode.limit;
+                } else {
+                    unsetLimit();
+                }
+            } else {
+                limit = inputNode.limit;
```

Comment:
If there is a limit demand pushed down, how to deal with it, can it be terminated in advance?

Reply:
```
select * from (     select *, rank() over (partition by v2 order by v3) as rk from t0 ) sub_t0 where rk < 5 limit 5;
```

Reply:
For the ordinary TopN, merge exchange will be performed at the receiver side, and the final size of the data is N (the limit value)

But for PartitionTopN, we cannot perform merge operation at the receiver side